From d85df25e77a80961fdb88304201b6fe8c1ae28c0 Mon Sep 17 00:00:00 2001
From: Dmitry Shachnev <mitya57@gmail.com>
Date: Thu, 9 Jan 2025 14:18:31 +0300
Subject: [PATCH] Disable ICU C++ header-only APIs, introduced in ICU 76

They require C++17 which we do not have, but we do not need them anyway.

Upstream-Status: Submitted [https://github.com/qtwebkit/qtwebkit/pull/1107]

Fixes:

qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/uset.h:1655:10: error: 'u16string_view' in namespace 'std' does not name a type; did you mean 'u16string'?
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/uset.h:1763:5: error: 'UnicodeString' does not name a type
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1553:25: error: 'enable_if_t' in namespace 'std' does not name a template type
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1553:36: error: expected '>' before '<' token
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1559:26: error: 'std::string_view' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1559:48: error: 'std::string_view' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1571:16: error: 'UnicodeString' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1571:35: error: 'UnicodeString' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1580:16: error: 'StringPiece' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1580:33: error: 'StringPiece' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1580:10: error: 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const' cannot be overloaded with 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1555:22: error: 'UnicodeString' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1555:57: error: 'UnicodeString' has not been declared
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1575:32: error: request for member 'getBuffer' in 'lhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1575:50: error: request for member 'length' in 'lhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1576:32: error: request for member 'getBuffer' in 'rhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1576:50: error: request for member 'length' in 'rhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1585:21: error: request for member 'data' in 'lhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1585:33: error: request for member 'length' in 'lhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1586:21: error: request for member 'data' in 'rhs', which is of non-class type 'int'
qtwebkit/5.15.16+git/recipe-sysroot/usr/include/unicode/ucol.h:1586:33: error: request for member 'length' in 'rhs', which is of non-class type 'int'
---
 Source/WTF/wtf/Platform.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 77999aaba..ea98d1a27 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -1142,6 +1142,7 @@
 /* FIXME: This does not belong in Platform.h and should instead be included in another mechanism (compiler option, prefix header, config.h, etc) */
 /* ICU configuration. Some of these match ICU defaults on some platforms, but we would like them consistently set everywhere we build WebKit. */
 #define U_SHOW_CPLUSPLUS_API 0
+#define U_SHOW_CPLUSPLUS_HEADER_API 0
 #ifdef __cplusplus
 #define UCHAR_TYPE char16_t
 #endif
